crypto/txscript/src/lib.rs

```diff
@@ -165,9 +213,18 @@ impl<'a, T: VerifiableTransaction, Reused: SigHashReusedValues> TxScriptEngine<'
             cond_stack: vec![],
             num_ops: 0,
             kip10_enabled,
+            runtime_sig_op_counting: None,
         }
     }
 
+    /// Returns the number of signature operations that were actually required during script execution.
+    /// Must only be called after script execution completes successfully.
+    ///
+    /// Returns the difference between the input's sig_op_limit and remaining sig ops.
+    pub fn sig_op_required(&self) -> Option<u8> {
```

Comment:
sig_op->sig_ops
Also, maybe `used_sig_ops` is a better name?

Reply:
2175dbe5481c4e9e4ff375b2c1d8c7c435e05bfc